### PR TITLE
fix(number-field): negative numbers handled

### PR DESCRIPTION
**Pull request by blunteshwar**
## Description

Number field was not correctly handling negative numbers when step was a floating point number

## Related issue(s)

-

## How has this been tested?

-   [ ] _Test case 1_
    1. Go here
    2. Do this
-   [ ] _Test case 2_
    1. Go here
    2. Do this

## Types of changes

-   [x] Bug fix (non-breaking change which fixes an issue)
-   [ ] New feature (non-breaking change which adds functionality)
-   [ ] Breaking change (fix or feature that would cause existing functionality to change)
-   [ ] Chore (minor updates related to the tooling or maintenance of the repository, does not impact compiled assets)

## Checklist

-   [x] I have signed the [Adobe Open Source CLA](http://opensource.adobe.com/cla.html).
-   [x] My code follows the code style of this project.
-   [x] If my change required a change to the documentation, I have updated the documentation in this pull request.
-   [x] I have read the **[CONTRIBUTING](<(https://github.com/adobe/spectrum-web-components/blob/main/CONTRIBUTING.md)>)** document.
-   [ ] I have added tests to cover my changes.
-   [x] All new and existing tests passed.
-   [x] I have reviewed at the Accessibility Practices for this feature, see: [Aria Practices](https://www.w3.org/TR/wai-aria-practices/)

## Best practices

This repository uses conventional commit syntax for each commit message; note that the GitHub UI does not use this by default so be cautious when accepting suggested changes. Avoid the "Update branch" button on the pull request and opt instead for rebasing your branch against `main`.

### CODE REVIEW

**Review by Westbrook — COMMENTED** [review]
This looks pretty good to me. If you could add a test to confirm that the original issue is now address this will be good to go!

File: packages/number-field/src/NumberField.ts [hunk 1]
```diff
@@ -489,6 +491,7 @@ export class NumberField extends TextfieldBase {
                 }
             }
         }
+        value *= signMultiplier;
```
Comment (Rajdeepc) [inline packages/number-field/src/NumberField.ts:494]: Is this a repetative statement? See [here](https://github.com/adobe/spectrum-web-components/pull/3673/files#diff-bbe60f2ecb5ae23472a9f446b13eaf5000919b731d68bee594c13a930d9b9933R468)

**Review by Rajdeepc — CHANGES_REQUESTED** [review]

File: packages/number-field/src/NumberField.ts (hunk 1 above)
Comment (blunteshwar) [reply packages/number-field/src/NumberField.ts:494]: yeah!
Basically I am checking if the the number is negative or not, if it is I am multiplying it by -1 to make it positive and the the function validateinput is doing its job and at the end before returning I am again multiplying by -1.

**Comment by Rajdeepc** [comment]
@blunteshwar if you are able to add a test we can merge this

**Review by Rajdeepc — COMMENTED** [review]
@blunteshwar if you are able to add a test we will be able to merge this

File: packages/number-field/src/NumberField.ts (hunk 1 above)
Comment (Westbrook) [reply packages/number-field/src/NumberField.ts:494]: It's an easy trap not to catch this, let's drop a comment in that explains this for future us.

Comment (blunteshwar) [reply packages/number-field/src/NumberField.ts:494]: okay!

**Comment by blunteshwar** [comment]
> @blunteshwar if you are able to add a test we will be able to merge this

done!

File: packages/number-field/src/NumberField.ts (hunk 1 above)
Comment (Westbrook) [reply packages/number-field/src/NumberField.ts:494]: Get this comment in and we're ready to merge!

Comment (blunteshwar) [reply packages/number-field/src/NumberField.ts:494]: done!

File: packages/number-field/src/NumberField.ts [hunk 2]
```diff
@@ -464,6 +464,8 @@ export class NumberField extends TextfieldBase {
     }
 
     private validateInput(value: number): number {
+        const signMultiplier = value < 0 ? -1 : 1;  // Basically I am checking if the the number is negative or not, if it is I am multiplying it by -1 to make it positive and the the function validateinput is doing its job and at the end before returning I am again multiplying by -1.
```
Comment (Rajdeepc) [inline packages/number-field/src/NumberField.ts:467]:
```suggestion
        const signMultiplier = value < 0 ? -1 : 1;  // Basically I am checking if the the number is negative or not, if it is I am multiplying it by -1 to make it positive and the the function validateinput is doing its job and at the end before returning I am again multiplying by -1.
 'signMultiplier' adjusts 'value' for 'validateInput' and reverts it before returning.
```

File: packages/number-field/src/NumberField.ts [hunk 3]
```diff
@@ -464,6 +464,9 @@ export class NumberField extends TextfieldBase {
...
+ 'signMultiplier' adjusts 'value' for 'validateInput' and reverts it before returning.
+        value *= signMultiplier;
```
Comment (Westbrook) [inline packages/number-field/src/NumberField.ts:469]: The comment updates got committed weirdly here.

Can you also update to `main` as @TarunAdobe's work was also in this file and there may be conflicts.

**Comment by Rajdeepc** [comment]
@Westbrook If we are good to go, let's land this 🛩️

**Review by Westbrook — APPROVED** [review]
:shipit:!

**Pull request merged**